librna/rnalib.c

```diff
@@ -1067,16 +1067,49 @@ int ss_energy(rsize i, rsize j) {
   return 0;
 }
 
+double get_mkpf_value(int x) {
+  /* -precalculate LOOKUP_SIZE mk_pf partition function values
+     -since input values can be negative,
+      values within the range -HALF <= x < HALF
+      will be calculated (HALF = LOOKUP_SIZE / 2) */
+
+  static bool init = false;
+  static const int HALF = LOOKUP_SIZE / 2;
+  static double lookup[LOOKUP_SIZE];
```

Comment:
Regarding the LOOKUP_SIZE (which is 10,000 right?): Is it a given that the array `lookup` has always at least 10k entries? Or could it be possible that it's smaller - leading to an "index out of bounds" error?

Reply:
Ah, never mind my question, I misunderstood the code.